docs/DeveloperGuide.md

```diff
@@ -2,21 +2,133 @@
 
 ## Acknowledgements
 
-{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links 
+to the original source as well}
 
 ## Design & implementation
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+The following is our class diagram for our whole project.
```

Comment:
An architecture diagram would be more suitable to describe the whole project.
This class diagram can be used to discuss how Customer class is being used in the project.

CliRental class seems to be very out of place here.
Do take note of the usage "Void" v.s. "void", "String" v.s. "string".
Do include multiplicity as well.
The class diagram seems to suggest aggregation instead of composition.